pkg/server/server.go

```diff
@@ -666,40 +674,47 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 
 	recvCh := make(chan *client.Packet, xfrChannelSize)
 
-	go s.serveRecvBackend(backend, stream, agentID, recvCh)
+	labels := runpprof.Labels(
+		"serverID", s.serverID,
+		"serverCount", strconv.Itoa(s.serverCount),
+		"agentID", agentID,
+	)
+	go runpprof.Do(context.Background(), labels, func(context.Context) { go s.serveRecvBackend(backend, stream, agentID, recvCh) })
 
 	defer func() {
 		klog.V(2).InfoS("Receive channel from agent is stopping", "agentID", agentID, "serverID", s.serverID)
 		close(recvCh)
 	}()
 
 	stopCh := make(chan error)
-	go func() {
-		for {
-			in, err := stream.Recv()
-			if err == io.EOF {
-				klog.V(2).InfoS("Receive stream from agent is closed", "agentID", agentID, "serverID", s.serverID)
-				close(stopCh)
-				return
-			}
-			if err != nil {
-				klog.ErrorS(err, "Receive stream from agent read failure")
-				stopCh <- err
-				close(stopCh)
-				return
-			}
+	go runpprof.Do(context.Background(), labels, func(context.Context) {
+		func() {
```

Comment:
Another candidate to move to a named function.

Reply:
Done